[#77] 검색: 검색결과에 따른 컬렉션뷰를 바인딩

gustn3965 opened the pull request:
## 개요
검색결과에 따른 컬렉션뷰를 바인딩

## 시연 
![검색결과](https://user-images.githubusercontent.com/48749182/136155958-ba299d92-b3c6-4e7a-aace-63b892926cd1.gif)


## 작업 사항

- 키보드위에 취소버튼 추가
    - `SearchTextField`에 키보드 나타날때 취소버튼 추가
- 검색결과화면 상단 레이블 추가 
    -  ![image](https://user-images.githubusercontent.com/48749182/136157214-9b0571eb-1581-45d9-9936-a6fafdc918ea.png)
    - `CategoryFilterView`를 이용하여 재사용함.
    - `CategoryFilterView`를 재사용하기 위해 약간의 리팩토링
    - `SearchResultsViewController`에 상단에 검색결과 레이블 뷰 추가
- 검색결과 화면 바인딩
    - `SearchViewController`에 검색완료 했을 떄 검색결과화면 나오도록 로직 구현
- 텍스트필드 `clearButton` 틴트
    - 에디팅모드일때 다크모드에서도 `clearButton` 색 변경

## 예외사항 
- 검색결과에서 모두보기 구현하지 않음 
- 모든 결과데이터는 테스트 데이터값

### Linked Issue

close #77 
close #84

gustn3965 committed:
[#77] ref: 키보드위에 취소버튼 추가
- SearchTextField에 키보드 나타날때 취소버튼 추가

gustn3965 committed:
[#77] ref: 검색결과화면 상단 레이블 추가
- CategoryFilterView를 이용하여 재사용함.
- CategoryFilterView를 재사용하기 위해 약간의 리팩토링
- SearchResultsViewController에 상단에 검색결과 레이블 뷰 추가

gustn3965 committed:
[#77] view: 검색결과 화면 바인딩
- SearchViewController에 검색완료 했을 떄 검색결과화면 나오도록 로직 구현

gustn3965 committed:
[#69] ref: 텍스트필드 clearButton 틴트
- 에디팅모드일때 다크모드에서도 clearButton 색 변경

jwonyLee commented on ThingLog/View/SearchTextField.swift:
```diff
@@ -168,6 +170,22 @@ extension SearchTextField {
+    func setupToolBar() {
+        let numberToolbar: UIToolbar = UIToolbar(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: 50))
+        numberToolbar.barStyle = .default
+        let action: UIBarButtonItem = UIBarButtonItem(title: "취소", style: .plain, target: self, action: #selector(cancleKeyboard))
```
:eyes: 지역 변수이긴 하지만... 이름이 `numberToolbar`인 이유가 있나요?_? 그냥 궁금해서..ㅎㅎ..

gustn3965 replied:
아뇨 ㅎㅎ 코드 붙여넣기 해서 이름을 제대로 변경안했네용 이것도 같이 반영할게요~!

jwonyLee commented on ThingLog/ViewController/SearchViewController.swift:
```diff
@@ -123,21 +141,59 @@ final class SearchViewController: UIViewController {
     
     func searchTextFieldShouldReturn(_ textField: UITextField) -> Bool {
         if recentSearchDataViewModel.isAutoSaveMode {
             recentSearchDataViewModel.add(textField.text!)
```
😎 공백만 입력되었을 때, 아무것도 입력하지 않고 검색했을 때도 `SearchResultsViewController`를 보여주고 있는데 이 부분도 필터링하는 부분이 있으면 좋을 거 같아요~ 공백 한 개일 때, 공백 n개일 때 다 따로따로 최근 검색에 저장되기도 하는데 이것도 같이 필터링되면 좋을 거 같아요! 나중에 수정하셔도 괜찮구여~

gustn3965 replied:
아 맞네요!! 좋은 지적 감사합니당! 반영할게요!

jwonyLee reviewed (approved):
LGTM :+1:

gustn3965 committed:
[#77] 지원님, 병연님 피드백 반영
- SearchTextField - toolBar 이름 변경
- SearchViewController - textField 엔터 조건 변경
- systemBlue 컬러 추가
- SearchTextField - toolBar 취소버튼 폰트, 색상 변경
- 최근검색어들 간격 사이 조정

gustn3965 commented:
지원님이 피드백 주신것과 병연님의 피드백을 반영했습니당!
